### ISSUE TITLE
Documentation for passing prefect token for terraform

### ISSUE THREAD

**Issue by brl0** [Quansight/qhub#661]
From my comment in Slack:

> from what I can tell, the PREFECT_TOKEN stopped getting passed to the terraform command line in GitHub Actions when the update was made to base the actions on qhub deploy rather than directly using terraform apply .  That change happened in PR#567.  Previously, the apply command was being called with these args: '-var="prefect_token=${{ secrets.PREFECT_TOKEN }}"' , so I think this will need to somehow be passed through to the terraform command line in qhub deploy , or at least something similar.

For reference, [here](https://github.com/Quansight/qhub/blob/caad7eaeb56b0a07cb4ecd1ea6c42352100f62be/qhub/template/%7B%7B%20cookiecutter.repo_directory%20%7D%7D/.github/workflows/jupyterhub.yaml#L89-L93) is the option being passed in qhub v0.3.11.

[Here](https://github.com/Quansight/qhub/blob/caad7eaeb56b0a07cb4ecd1ea6c42352100f62be/qhub/deploy.py#L113) is the apply command that would result in prefect being deployed in the latest version, which calls [this](https://github.com/Quansight/qhub/blob/caad7eaeb56b0a07cb4ecd1ea6c42352100f62be/qhub/provider/terraform.py#L80) function.

cc @iameskild, @kcpevey, @aktech

**Comment by tylerpotts** [comment]
per @aktech This will work if an environment is set `TF_VAR_PREFECT_TOKEN=<prefect_token>`. See https://www.terraform.io/docs/language/values/variables.html#environment-variables for more info.

We're re-classifying this as a documentation issue, since secrets ideally shouldn't be in plain text in `qhub-config.yaml`

**Comment by brl0** [comment]
Thanks, I did not know about `TF_VAR_`, that's very helpful and great to know.

Reading another section of the linked document about [tfvars](https://www.terraform.io/docs/language/values/variables.html#variable-definitions-tfvars-files) files, that might be another simple and easy to manage way to manage passing secrets, at least for the initial deployment, although they shouldn't be needed afterwards since deployment is handled by CICD which will manage secrets separately.

However, I noticed that we currently don't exclude those in the `.gitignore`, but based on other examples [here](https://github.com/github/gitignore/blob/master/Terraform.gitignore) and [here](https://github.com/gruntwork-io/intro-to-terraform/blob/master/.gitignore) it looks like we probably should.  Here is a comment from the first link:

```
# Exclude all .tfvars files, which are likely to contain sentitive data, such as
# password, private keys, and other secrets. These should not be part of version 
# control as they are data points which are potentially sensitive and subject 
# to change depending on the environment.
```

So I think maybe that should be changed, probably whenever the documentation is updated to reflect ways of passing variables into terraform.

As always, thanks for the guidance and support!

**Comment by aktech** [comment]
Fixed here: #641

**Comment by aktech** [comment]
@brl0 Just saw your comments regarding adding `*.tfvars`. It seems I missed it earlier. I'll add that in a follow up PR
